src/scripts/services/notifications/dimItemNotificationGenerator.factory.js

```diff
@@ -0,0 +1,83 @@
+const angular = require('angular');
+const _ = require('underscore');
+
+function shuffle(array) {
+  var currentIndex = array.length, temporaryValue, randomIndex;
+
+  // While there remain elements to shuffle...
+  while (currentIndex !== 0) {
+
+    // Pick a remaining element...
+    randomIndex = Math.floor(Math.random() * currentIndex);
+    currentIndex -= 1;
+
+    // And swap it with the current element.
+    temporaryValue = array[currentIndex];
+    array[currentIndex] = array[randomIndex];
+    array[randomIndex] = temporaryValue;
+  }
+
+  return array;
+}
+
+(function() {
+  'use strict';
+
+  angular.module('dimApp')
+    .factory('dimItemNotificationGenerator', ItemNotificationGenerator)
+    .run(['dimItemNotificationGenerator', (dimItemNotificationGenerator) => {
+      // Assuming on for prototype
+      dimItemNotificationGenerator.start();
+    }]);
+
+  function ItemNotificationGenerator($rootScope, dimNotificationsService) {
+    const createNotification = dimNotificationsService.registerNotificationGenerator({
+      name: "Item Notification Generator",
+      description: "I tell people when they get shiny new loot."
+    });
+    let cancelReloadListener;
+    const alreadyNotified = [];
+
+    // Ideally we'd start/stop this as the user turns it on/off to avoid unneeded processing.
+    return {
+      active: false,
+      start: () => {
+        var self = this;
+        function checkForNewStuff(store) {
+          if (!store) {
+            return;
+          }
+          const newItems = shuffle(_.filter(store.items, function(item) { return !item.notransfer && item.equipment && !item.classified && _.indexOf(alreadyNotified, item.id) < 0; }));
```

Comment:
The shuffle exists to make demoing the prototype a little easier and a little more variable. When you receive the notification, it notes that you received "{Item name} and x other items." The item chosen for the notification is the first in the list, allowing you to see a different image on each app load.

The "new" field was my first choice, but this "new" value does not clear on each inventory refresh; an item is considered new if it was obtained during the current session. Therefore, you'll be notified about the same items on each inventory refresh.